### DISCUSSION TITLE
파이캐럿 실험내용

### DISCUSSION THREAD

**Discussion by alinghi** [bcaitech1/p4-dkt-lastffang#99] [Ideas]
Train을 StratifiedKFold로 10번 나누어서 실험, Time Series 무시 + 순수 feature로만 learning

## 최적의 알고리즘 모델 ML 후보

ID | Name
-- | --
‘lr’ | Logistic Regression
‘knn’ | K Nearest Neighbour
‘nb’ | Naives Bayes
‘dt’ | Decision Tree Classifier
‘svm’ | SVM – Linear Kernel
‘rbfsvm’ | SVM – Radial Kernel
‘gpc’ | Gaussian Process Classifier
‘mlp’ | Multi Level Perceptron
‘ridge’ | Ridge Classifier
‘rf’ | Random Forest Classifier
‘qda’ | Quadratic Discriminant Analysis
‘ada’ | Ada Boost Classifier
‘gbc’ | Gradient Boosting Classifier
‘lda’ | Linear Discriminant Analysis
‘et’ | Extra Trees Classifier
‘xgboost’ | Extreme Gradient Boosting
‘lightgbm’ | Light Gradient Boosting
‘catboost’ | CatBoost Classifier

## 결과
[LGBMClassifier(boosting_type='gbdt', class_weight=None, colsample_bytree=1.0,
                importance_type='split', learning_rate=0.1, max_depth=-1,
                min_child_samples=20, min_child_weight=0.001, min_split_gain=0.0,
                n_estimators=100, n_jobs=-1, num_leaves=31, objective=None,
                random_state=6396, reg_alpha=0.0, reg_lambda=0.0, silent=True,
                subsample=1.0, subsample_for_bin=200000, subsample_freq=0),
 GradientBoostingClassifier(ccp_alpha=0.0, criterion='friedman_mse', init=None,
                            learning_rate=0.1, loss='deviance', max_depth=3,
                            max_features=None, max_leaf_nodes=None,
                            min_impurity_decrease=0.0, min_impurity_split=None,
                            min_samples_leaf=1, min_samples_split=2,
                            min_weight_fraction_leaf=0.0, n_estimators=100,
                            n_iter_no_change=None, presort='deprecated',
                            random_state=6396, subsample=1.0, tol=0.0001,
                            validation_fraction=0.1, verbose=0,
                            warm_start=False),
 AdaBoostClassifier(algorithm='SAMME.R', base_estimator=None, learning_rate=1.0,
                    n_estimators=50, random_state=6396),
 KNeighborsClassifier(algorithm='auto', leaf_size=30, metric='minkowski',
                      metric_params=None, n_jobs=-1, n_neighbors=5, p=2,
                      weights='uniform'),
 LinearDiscriminantAnalysis(n_components=None, priors=None, shrinkage=None,
                            solver='svd', store_covariance=False, tol=0.0001)]

## 상위 5개 모델(AUC 기준) 
**LGBM, GradientBoost, AdaBoost, KNN, LinearDiscriminant Analysis**

중간에 터져서 결과만 저장되어 있고, 상세한 수치는 날라갔어요....

KNN이 Accuracy 0.7114, AUC 0.7056인거만 남아 있어요

## 이후 튜닝
- LGBM 튜닝 모델 Accuracy 0.7567 AUC 0.7916
